server/s3_copy_object_action.cc

```diff
@@ -181,9 +328,20 @@ void S3CopyObjectAction::send_response_to_s3_client() {
       (is_error_state() && !get_s3_error_code().empty())) {
     // Metadata saved for object is always a success condition.
     assert(s3_copy_action_state != S3CopyObjectActionState::metadataSaved);
+    S3Error error(get_s3_error_code(), request->get_request_id());
+
+    if (S3CopyObjectActionState::validationFailed == s3_copy_action_state &&
```

Comment:
We cannot have this state at all right, in validation itself we will bail out right

Reply:
yes, in validate_copyobject_request() function, I set the state as "validationFailed", s3 error code to "InvalidRequest." and call send_response_to_s3_client().

I have added this check here to see if s3 error code is "InvalidRequest", so set the error message accordingly.